lib/kafka/protocol/describe_groups_response.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Kafka
+  module Protocol
+    class DescribeGroupsResponse
+      class Group
+        attr_reader :error_code, :group_id, :state, :members, :protocol
+
+        def initialize(error_code:, group_id:, protocol_type:, protocol:, state:, members:)
+          @error_code = error_code
+          @group_id = group_id
+          @protocol_type = protocol_type
+          @protocol = protocol
+          @state = state
+          @members = members
+        end
+      end
+
+      class Member
+        attr_reader :member_id, :client_id, :client_host, :member_assignment
+
+        def initialize(member_id:, client_id:, client_host:, member_assignment:)
+          @member_id = member_id
+          @client_id = client_id
+          @client_host = client_host
+          @member_assignment = member_assignment
+        end
+      end
+
+      attr_reader :error_code, :groups
+
+      def initialize(groups:)
+        @groups = groups
+      end
+
+      def self.decode(decoder)
+        groups = decoder.array do
+          error_code = decoder.int16
+          group_id = decoder.string
+          state = decoder.string
+          protocol_type = decoder.string
+          protocol = decoder.string
+
+          members = decoder.array do
+            member_id = decoder.string
+            client_id = decoder.string
+            client_host = decoder.string
+            metadata = decoder.bytes
```

Comment:
This might contain some useful information, but I couldn't find how to decode it in the [protocol docs](https://kafka.apache.org/protocol) (Unless I missed something)

> The metadata corresponding to the current group protocol in use (will only be present if the group is stable).

Reply:
@jasondoc3 The group protocol is the metadata information that the group agree on at the Join Group step. In that step, each consumer group member proposes a list of protocol and its metadata that all members in the group could agree on. The group leader picks a protocol that all members in the group supports and store that information. Therefore, this field is defined by the client. You don't need to parse this. `ruby-kafka` gem doesn't support a custom group protocol. 

P/s: If you are curious, you can look at `lib/kafka/protocol/join_group_request.rb` and `lib/kafka/protocol/consumer_group_protocol.rb`